inspirehep/modules/workflows/views.py

```diff
@@ -376,3 +401,180 @@ def robotupload_callback():
         responses[recid] = response
 
     return jsonify(responses)
+
+
+def _validate_workflow_payload(keys):
+    """Validate and get the workflow data from the request.
+
+    It will first check if all the given keys are part of
+    the request's payload, after will validate the ``metadata`` against
+    the ``hep`` JSONSchema and finally it will pass the ``workflow_data``
+    kwarg.
+
+    Args:
+        keys (list(str)): the keys to check if they exist.
+
+    Raises:
+        CallbackMalformedError: if any of the ``keys`` is missing from
+            the request payload.
+        CallbackValidationError: if the workflow ``metadata`` is not valid
+            against ``hep`` JSONSchema.
+    """
+    workflow_data = request.get_json()
```

Comment:
it's probably better to pass the json as a parameter, so this function does not need to know about the request thingie.

Reply:
The idea is to avoid using flask in the inner functions as much as possible, leaving the http interfacing for the functions that are directly called for it (that way also makes this one easier to test by itself).